athenian/database_pg.go

```diff
@@ -223,6 +235,104 @@ func (db *database) AthenianToGithub(ctx context.Context, id AccountID) ([]Githu
 	return out, rows.Err()
 }
 
+func (db *database) GetInstaflowStatus(ctx context.Context, accID AccountID) (*InstaflowStatus, error) {
+	return db.getInstaflowStatus(ctx, accID)
+}
+
+func (db *database) getInstaflowStatus(ctx context.Context, accID AccountID) (*InstaflowStatus, error) {
+	row := db.db.QueryRow(
+		ctx,
+		`SELECT account_id, account_created, fetch_started, fetch_completed, consistency_started, consistency_completed, precompute_started, precompute_completed, current_status
+		FROM public.installation_progress WHERE account_id = $1`,
+		int64(accID),
+	)
+
+	instaflowStatus, err := scanInstaflowStatus(row)
+	if err != nil {
+		return nil, err
+	}
+	return &instaflowStatus, err
+}
+
+func scanInstaflowStatus(sc dbs.Scanner) (InstaflowStatus, error) {
+	var (
+		accID                int64
+		accountCreated       *time.Time
+		fetchStarted         *time.Time
+		fetchCompleted       *time.Time
+		consistencyStarted   *time.Time
+		consistencyCompleted *time.Time
+		precomputeStarted    *time.Time
+		precomputeCompleted  *time.Time
+		status               string
+	)
+	err := sc.Scan(&accID, &accountCreated, &fetchStarted, &fetchCompleted, &consistencyStarted, &consistencyCompleted, &precomputeStarted, &precomputeCompleted, &status)
+	if err == pgx.ErrNoRows {
+		err = ErrNotFound
+	}
+
+	if err != nil {
+		return InstaflowStatus{}, err
+	}
+
+	return InstaflowStatus{
+		AccID:                AccountID(accID),
+		AccountCreated:       accountCreated.UTC(),
+		FetchStarted:         fetchStarted.UTC(),
+		FetchCompleted:       fetchCompleted.UTC(),
+		ConsistencyStarted:   consistencyStarted.UTC(),
+		ConsistencyCompleted: consistencyCompleted.UTC(),
+		PrecomputeStarted:    precomputeStarted.UTC(),
+		PrecomputeCompleted:  precomputeCompleted.UTC(),
+		Status:               InstallStatus(status),
+	}, err
+}
+
+func (db *database) UpdateInstaflowStatus(ctx context.Context, accID AccountID, timestamp time.Time, status InstallStatus) error {
+	tx, err := db.db.Begin(ctx)
+	if err != nil {
+		return err
+	}
+	defer tx.Rollback(ctx)
+
+	tsColumn := InstallStatusTimestampColumns[status]
+
+	var qu string
+
+	ext, err := db.getInstaflowStatus(ctx, accID)
+	if err != nil && err != ErrNotFound {
+		return err
+	} else if err != ErrNotFound {
```

Comment:
I'm really confused by this. Shouldn't it be `err == ErrNotFound`?

Reply:
ohh. right. my bad.